native/SampleApps/SmartSyncExplorer/RecentContactsTodayExtension/TodayViewController.m

```diff
@@ -0,0 +1,129 @@
+ DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+ WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY
+ WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+#import "TodayViewController.h"
+
+#import <SmartSyncExplorerCommon/SmartSyncExplorerCommon.h>
+#import <NotificationCenter/NotificationCenter.h>
+#import <SmartStore/SalesforceSDKManagerWithSmartStore.h>
+#import <SalesforceSDKCore/SFSDKDatasharingHelper.h>
+#import <SalesforceSDKCore/NSUserDefaults+SFAdditions.h>
+#import <SalesforceSDKCore/SFRestRequest.h>
+#import <SalesforceSDKCore/SFRestAPI.h>
+#import <SalesforceSDKCore/SFLoggerMacros.h>
+#import <SalesforceSDKCore/SFLogger.h>
+#import <SmartStore/SFQuerySpec.h>
+
+static int const kNumberOfRecords = 3;
+
+@interface TodayViewController () <NCWidgetProviding,UITableViewDelegate,UITableViewDataSource>{
+      NSMutableArray *_contacts;
+}
+@property (weak, nonatomic) IBOutlet UITableView *todayTableView;
+@property (nonatomic, strong) SObjectDataManager *dataMgr;
+
+@end
+
+@implementation TodayViewController
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    #if defined(DEBUG)
+        [SFLogger sharedLogger].logLevel = SFLogLevelDebug;
+    #else
+        [SFLogger sharedLogger].logLevel = SFLogLevelInfo;
+    #endif
+
+    _contacts = [NSMutableArray new];
+    [self.todayTableView setDataSource:self];
+    [self.todayTableView setDelegate:self];
+}
+
+- (void) refreshList {
+    [self.todayTableView reloadData];
+}
+
+- (void)didReceiveMemoryWarning {
+    [super didReceiveMemoryWarning];
+    // Dispose of any resources that can be recreated.
+}
+
+- (void)widgetPerformUpdateWithCompletionHandler:(void (^)(NCUpdateResult))completionHandler {
+    
+    SmartSyncExplorerConfig *config = [SmartSyncExplorerConfig sharedInstance];
+
+    [SFSDKDatasharingHelper sharedInstance].appGroupName = config.appGroupName;
+    [SFSDKDatasharingHelper sharedInstance].appGroupEnabled = YES;
+    if( [self userLoginStatus] ) {
+        [self log:SFLogLevelError format:@"User has logged in"];
+        [SalesforceSDKManager setInstanceClass:[SalesforceSDKManagerWithSmartStore class]];
+        [SalesforceSDKManager sharedManager].connectedAppId = config.remoteAccessConsumerKey;
+        [SalesforceSDKManager sharedManager].connectedAppCallbackUri = config.oauthRedirectURI;
+        [SalesforceSDKManager sharedManager].authScopes = config.oauthScopes;
+        [SalesforceSDKManager sharedManager].authenticateAtLaunch = config.appGroupsEnabled;
+        
+        NSArray *accounts = [[SFUserAccountManager sharedInstance] allUserAccounts];
+        if(accounts && accounts.count>0) {
+            // check for actual Account or id of account or pick the first one?
```

Comment:
How about:
```obj-c
SFUserAccountIdentity *activeUserIdentity = [SFUserAccountManager sharedInstance].activeUserIdentity;
SFUserAccount *currentUser = [[SFUserAccountManager sharedInstance] userAccountForUserIdentity:activeUserIdentity];
[SFUserAccountManager sharedInstance].currentUser = currentUser;
```
Active user identity is rooted in `NSUserDefaults`, so it should be persistent between apps.